### PR TITLE
LIME-9: Add checkbox

### LINKED ISSUE #9: Create Checkbox component
Create a reusable Checkbox component that integrates with react-hook-form.
Image
[Design](https://www.figma.com/file/RRE9g9LIo3a1AK89HBlN6U/BSA_Lime?type=design&node-id=0-1&mode=design&t=kbzEh3yBFKNMZCBA-0)

### PR DESCRIPTION
**Pull request by DavidCharif**
Added reusable checkbox component

### CODE REVIEW

File: frontend/src/libs/components/checkbox/checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    label: string;
+    control: Control<T>;
+};
+
+const Checkbox = <T extends FieldValues>({
+    name,
+    label,
+    control,
+}: CheckboxProperties<T>): ReactElement => {
+    const { field } = useController<T>({ name, control });
```
Comment (ArtemLymarenko) [inline frontend/src/libs/components/checkbox/checkbox.tsx]: You can use useFormController that's already in the hooks folder. Path: bundles -> common -> hooks

**Comment by ArtemLymarenko** [comment]
Link an issue to your PR.

**Review by ArtemLymarenko — APPROVED** [review]

File: frontend/src/libs/components/checkbox/checkbox.tsx
Comment (stpalamar) [inline frontend/src/libs/components/checkbox/checkbox.tsx:1]: checkbox component must be in 
**src >> bundles >> common >> components >> checkbox**

Comment (DavidCharif) [reply frontend/src/libs/components/checkbox/checkbox.tsx:1]: image
Right! 
Thanks

**Review by anton-otroshchenko — APPROVED** [review]

**Review by jhladkov — APPROVED** [review]

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
we dont have error state

**Comment by DavidCharif** [comment]
image
image

File: frontend/src/bundles/common/components/checkbox/checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            <label
+                className={getValidClassNames(
+                    field.value ? 'text-lm-grey-200' : 'text-lm-grey-500',
+                    'cursor-pointer',
+                )}
+                htmlFor="toggle-checkbox"
+            >
+                {label}
+                {hasError && (
+                    <span className="text-lm-red">{error as string}</span>
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/checkbox/checkbox.tsx]: looks like we should show an error under the component 
image

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/checkbox/checkbox.tsx (hunk 2 above)
Comment (DavidCharif) [reply frontend/src/bundles/common/components/checkbox/checkbox.tsx]: image
Like this?

File: frontend/src/bundles/common/components/icons/check-icon/check-icon.tsx
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/icons/check-icon/check-icon.tsx:1]: check-icon.tsx

File: frontend/src/bundles/common/components/icons/check-icon/check-icon.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { CheckIcon } from '@heroicons/react/16/solid';
+
+type CheckProperties = {
+    className?: string;
+};
+
+const CheckIconComponent = ({ className }: CheckProperties): JSX.Element => {
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/icons/check-icon/check-icon.tsx]:
```suggestion
const CheckIcon = ({ className }: CheckProperties): JSX.Element => {
```

Comment (DavidCharif) [reply frontend/src/bundles/common/components/icons/check-icon/check-icon.tsx]: image
image

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**